src/Microsoft.Health.Fhir.Core/Features/Search/SearchHelperUtilities.cs

```diff
@@ -3,39 +3,52 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
+using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 using EnsureThat;
-using Microsoft.Health.Core.Extensions;
-using Microsoft.Health.Fhir.Core.Features.Search.Registry;
+using Microsoft.Health.Fhir.Core.Models;
 
 namespace Microsoft.Health.Fhir.Core.Features.Search
 {
     public static class SearchHelperUtilities
     {
         /// <summary>
-        /// Given a list of <see cref="ResourceSearchParameterStatus"/> calculates a hash using the
-        /// <see cref="ResourceSearchParameterStatus.Uri"/> and <see cref="ResourceSearchParameterStatus.LastUpdated"/>
-        /// values of each component. The same collection of search parameter status (irrespective of their order in the input)
+        /// Given a list of <see cref="SearchParameterInfo"/> calculates a hash using the
+        /// <see cref="SearchParameterInfo.Url"/>, <see cref="SearchParameterInfo.Type"/>,
+        /// <see cref="SearchParameterInfo.Expression"/>, <see cref="SearchParameterInfo.TargetResourceTypes"/>, and
+        /// <see cref="SearchParameterInfo.BaseResourceTypes"/>,
+        /// values of each component. The same collection of search parameter infos (irrespective of their order in the input)
         /// will return the same hash.
         /// </summary>
-        /// <param name="resourceSearchParameterStatus">A list of <see cref="ResourceSearchParameterStatus" /></param>
+        /// <param name="searchParamaterInfos">A list of <see cref="SearchParameterInfo" /></param>
         /// <returns>A hash based on the search parameter uri and last updated value.</returns>
-        public static string CalculateSearchParameterHash(IEnumerable<ResourceSearchParameterStatus> resourceSearchParameterStatus)
+        public static string CalculateSearchParameterHash(IEnumerable<SearchParameterInfo> searchParamaterInfos)
         {
-            EnsureArg.IsNotNull(resourceSearchParameterStatus, nameof(resourceSearchParameterStatus));
-            EnsureArg.IsGt(resourceSearchParameterStatus.Count(), 0, nameof(resourceSearchParameterStatus));
+            EnsureArg.IsNotNull(searchParamaterInfos, nameof(searchParamaterInfos));
+            EnsureArg.IsGt(searchParamaterInfos.Count(), 0, nameof(searchParamaterInfos));
 
-            StringBuilder sb = new StringBuilder();
-            foreach (ResourceSearchParameterStatus searchParameterStatus in resourceSearchParameterStatus.OrderBy(x => x.Uri.ToString()))
+            HashCode hashCode = default;
```

Comment:
>HashCode [](start = 12, length = 8)

Just want to note what by using HashCode object you heavily reduce dimensionality of hash (it was 256 bit, now it would be 32).
Which is probably fine, considering low amount of objects we need to hash.

Reply:
Yes. When using this kind of hash code, it's usually followed up by a full equality check, which we are not doing. It seems like we should be doing a proper cryptographic hash here.

Reply:
Also, it we are persisting these hashes, then this approach is not deterministic

Reply:
Reverted the changes to the hash calculation, because as John mentioned the new approach would be deterministic.